### PR TITLE
[API] Upgrade CatalogPromotion resources to APIP 3

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | api-platform-3
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Channel.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                <value name="groups">
+                    <values>
+                        <value>sylius</value>
+                    </values>
+                </value>
+            </values>
+        </validationContext>
+
+        <operations>
+            <operation class="ApiPlatform\Metadata\Get" uriTemplate="admin/channels/{code}.{_format}">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Channel.xml]:
```suggestion
            <operation class="ApiPlatform\Metadata\Get" uriTemplate="/admin/channels/{code}.{_format}">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotionTranslation.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                <value name="groups">
+                    <values>
+                        <value>sylius</value>
+                    </values>
+                </value>
+            </values>
+        </validationContext>
+
+        <operations>
+            <operation class="ApiPlatform\Metadata\Get" uriTemplate="admin/catalog-promotion-translations/{id}.{_format}">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotionTranslation.xml]:
```suggestion
            <operation class="ApiPlatform\Metadata\Get" uriTemplate="/admin/catalog-promotion-translations/{id}.{_format}">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                <value name="groups">
+                    <values>
+                        <value>sylius</value>
+                    </values>
+                </value>
+            </values>
+        </validationContext>
+
+        <operations>
+            <operation class="ApiPlatform\Metadata\Get"  uriTemplate="/admin/catalog-promotions/{code}.{_format}">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml]: New uriTemplate style `{code}.{_format}` wasn't in use in previous APIP3 PRs, please tell me where can I read about it?

I cannot find mentioned
https://api-platform.com/docs/core/content-negotiation/

Will `/admin/catalog-promotions/some-code` work or I will need to specify format each time, like `/admin/catalog-promotions/some-code.jsonld`?

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+                <filters>
+                    <filter>sylius.api.catalog_promotion_search_filter</filter>
+                    <filter>sylius.api.catalog_promotion_enabled_filter</filter>
+                    <filter>sylius.api.catalog_promotion_start_date_filter</filter>
+                    <filter>sylius.api.catalog_promotion_end_date_filter</filter>
+                    <filter>Sylius\Bundle\ApiBundle\Filter\Doctrine\CatalogPromotionChannelFilter</filter>
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml:58]: Would be cool to unify services definitions (I assume it's services) either to string or FQCN.
```suggestion
                    <filter>sylius.api.catalog_promotion_channel_filter</filter>
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 5]
```diff
@@ -29,7 +29,9 @@
         <import resource="services/payments.xml" />
         <import resource="services/providers.xml" />
         <import resource="services/state_processors.xml" />
+        <import resource="services/serializers.xml" />
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml]: Would be cool to sort alphabetically

```xml
<import resource="services/section_resolver.xml" />
<import resource="services/serializers.xml" />
<import resource="services/state_processors.xml" />
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml (hunk 3 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml]: Generally, without using `_format` in the browser, I receive html (which is the default). However, if I add `_format`, I can display the response in `jsonld` by using  `/admin/catalog-promotions/some-code.jsonld`. Similary, for `json`, I can use `.json`.  I saw this in SymfonyCasts and [here](https://api-platform.com/docs/core/subresources/)

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml (hunk 4 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml:58]: Filters created in Sylius have an ID as FQCN, while those using filters from APIP have a string. We will see what others have to say.

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml (hunk 3 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml]: We decided within the team that we're doing it without `_format`

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>shop:catalog_promotion:read</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+            </operation>
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml:186]:
```suggestion
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**